### PR TITLE
Remove underscore limitation for web ui route functions

### PR DESCRIPTION
**Pull request by msmcconnell**
# PR Details
## Description

This PR allows route files to be named in any fashion by removing the checks for underscores and spaces from the web ui. 

## Related Issue
usdot-fhwa-stol/CARMAPlatform#407

## Motivation and Context

Prevent a source of error for developers where they misname their route files
## How Has This Been Tested?

I have tested this on the pacifica using carma config edit to change only the active web ui image. The rest of the build was from Urus. 

I tested it with route file containing underscores in the name. 
## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](Contributing.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: website/scripts/control.js [hunk 1]
```diff
@@ -107,14 +107,11 @@ function createRadioElement(container, radioId, radioTitle, itemCount, groupName
     newInput.type = 'radio';
     newInput.name = groupName;
 
-    //TODO: Remove this when RouteID has been changed to have no spaces.
-    //Currently, RouteID and RouteName are same and have spaces, but ID should not have any spaces. For now, updating to have underscore
-    var revisedId = radioId.toString().replace(/ /g, '_');
```
Comment (maefromm) [inline website/scripts/control.js]: So Mike after re-reading my comment above, the replace command is to resolve the issue when the filename has spaces. I am using the filename as the ID. So the real fix would be on Route.msg to have RouteID to be numerical or not have spaces, and maybe underscores/dashes/special characters. And have Route Name be the actual file name with underscores/dashses/spaces/special characters (if allowed in Linux filenaming). IDs should be unique so I'd go with numerical value. 

Once that's fixed then your fixes here should work. 

HEre's the route.msg format. Currently route_generator updates routeID and routeName with the filename. 
# The route id
string routeID

# The route name for display to users
string routeName

File: website/scripts/rosbridge.js [hunk 2]
```diff
@@ -420,11 +420,9 @@ function setRoute(id) {
...
 
-    //TODO: Remove this when Route Manager has updated the RouteID to not have spaces. For now have to do this.
-    var selectedRouteid = id.toString().replace('rb', '').replace(/_/g, ' ');
+    var selectedRouteid = id.toString().replace('rb', '');
 
-    // Then we create a Service Request.
-    // replace rb with empty string and underscore with space to go back to original ID from topic.
+    // Create a Service Request.
     var request = new ROSLIB.ServiceRequest({
         routeID: selectedRouteid
```
Comment (maefromm) [inline website/scripts/rosbridge.js]: If you're removing selectedRouteid, this should pass in "id".

File: website/scripts/control.js (hunk 1 above)
Comment (maefromm) [reply website/scripts/control.js]: @msmcconnell @qswawrq another thought is to remove routeID in route.msg, and just have routeName be the filename in the new routeNode. This is under the assumption we don't really use routeID anywhere else besides route_generator. 

If you agree that we don't really use RouteID anywhere else, then for now, I would say update the UI to automatically generate or re-generate the radioID using numerical values and have the radioTitle to be the routeFileName (or routeName). Then I don't think you will need to make changes to the route_generator.

Comment (maefromm) [reply website/scripts/control.js]: Nevermind, I see RouteID getting referenced in RouteState.msg and SetActiveRoute.msg, which would impact guidance I assume. So I think just go back to my original suggestion above. Also verified in our old repo, I see comment in the RouteWorker.java in the old repo (line229) saying ".setRouteID(route.getRouteName()); //TODO come up with better method of defining the route id". And that's where my comment in the UI matches that it expects routeID and routeName to be different.

**Comment by msmcconnell** [comment]
@maefromm Per our discussion over the phone, this change will work for the current use case. However, after usdot-fhwa-stol/CARMAPlatform/tree/integration/routing is merged in we will need to re-verify that the UI is still compatable and the system as a whole supports enhancement request 
usdot-fhwa-stol/CARMAPlatform#549

**Comment by maefromm** [comment]
@qswawrq @msmcconnell I've approved but please merge after it's been tested using the following: 

tFHRC.csv
Tfhrc.csv
TFHRC_1234.csv
TFHRC-1234.csv
NOTE: Per Mike, space like "TFHRC Circle" is no longer allowed.

**Comment by msmcconnell** [comment]
We verified the UI and route_generator communicate properly for each of those test cases. I created
https://github.com/usdot-fhwa-stol/CARMAPlatform/pull/550

To provide a warning if someone tries to load a route with spaces in the name.

**Pull request merged**